### PR TITLE
Remove BLS signatures aggregate

### PR DESCRIPTION
**Pull request by K1li4nL**
As the BLS signature aggregate scheme is vulnerable to rogue public-key attack, this pr removes the "aggregate" part of the code and use BDN where BLS aggregates were used.

### CODE REVIEW

File: pairing/bls12381/bls12381_test.go [hunk 1]
```diff
@@ -730,31 +733,47 @@ func BLSBenchmark(b *testing.B, curveOption string) {
...
+	}
 	for _, n := range numSigs {
+
 		// Benchmark aggregation of public keys
 		b.Run(fmt.Sprintf("AggregatePublicKeys-G1 on %d signs", n), func(bb *testing.B) {
 			for j := 0; j < bb.N; j++ {
-				result = schemeOnG1.AggregatePublicKeys(pubKeysOnG1[:n]...)
+				result, err = schemeOnG1.AggregatePublicKeys(maskG1)
+				if err != nil {
+					panic(err)
```
Comment (pierluca) [inline pairing/bls12381/bls12381_test.go:752]: why panic instead of a `require.NoError`?

**Review by pierluca — COMMENTED** [review]
Overalls LGTM, but this could benefit from @AnomalRoil 's review.

File: pairing/bls12381/bls12381_test.go (hunk 1 above)
Comment (K1li4nL) [reply pairing/bls12381/bls12381_test.go:752]: No good reason apparently, I didn't expect `*time.T` and `*time.B` to be swappable, but it seems to work that way :+1:

**Comment by K1li4nL** [comment]
Turning this back into draft, I didn't properly ran the benchmarks... It seems that our bdn implementation assumes kyber.Scalar to be mod.Int, making it incompatible to use with circl_bls12381

File: sign/bls/bls.go [hunk 2]
```diff
@@ -2,7 +2,7 @@
 // was introduced in the paper "Short Signatures from the Weil Pairing". BLS
 // requires pairing-based cryptography.
 //
-// Deprecated: This version is vulnerable to rogue public-key attack and the
+// This version is vulnerable to rogue public-key attack and the
```
Comment (AnomalRoil) [inline sign/bls/bls.go:5]: Maybe specify it's in case of aggregation only and that regular BLS signatures are fine

File: sign/bdn/bdn.go [hunk 3]
```diff
@@ -61,14 +61,39 @@ func hashPointToR(pubs []kyber.Point) ([]kyber.Scalar, error) {
...
+		b, err := mod.NewIntBytes(out[i*16:(i+1)*16], modulus128, kyber.LittleEndian).MarshalBinary()
+		if err != nil {
+			return nil, err
+		}
+		if g.Scalar().ByteOrder() == kyber.BigEndian {
+			reverse(b, b)
+		}
+
+		coefs[i] = g.Scalar()
+		coefs[i].SetBytes(b)
```
Comment (AnomalRoil) [inline sign/bdn/bdn.go:73]: `g.Scalar` is of size 32 whereas the coefs in `bdn` are meant to be of size 128 bits, e.g. 16.
I think that's explaining the issues you've had so far.

Also why not just:

```suggestion
		b, err := mod.NewIntBytes(out[i*16:(i+1)*16], modulus128, g.Scalar().ByteOrder()).MarshalBinary()
		if err != nil {
			return nil, err
		}

		coefs[i] = b
```

Comment (K1li4nL) [reply sign/bdn/bdn.go:73]: Although it looks cleaner, this is a breaking change, these bytes in `out` were always interpreted in little-endian mod 2^128 with no regard to the endianness of the suite.

File: sign/bdn/bdn.go [hunk 4]
```diff
@@ -61,14 +61,39 @@ func hashPointToR(pubs []kyber.Point) ([]kyber.Scalar, error) {
...
 	}
 
 	return coefs, nil
 }
 
+// reverse copies src into dst in byte-reversed order and returns dst,
+// such that src[0] goes into dst[len-1] and vice versa.
+// dst and src may be the same slice but otherwise must not overlap.
+func reverse(dst, src []byte) []byte {
+	if dst == nil {
```
[the remark this answers is not in the document]
Comment (K1li4nL) [reply sign/bdn/bdn.go:83]: good point, I enforced the equality, there and where I copied this function in `mod/int`

File: sign/bdn/bdn.go (hunk 3 above)
Comment (AnomalRoil) [reply sign/bdn/bdn.go]: I think this is getting changed in #553 anyway now, no?

Comment (K1li4nL) [reply sign/bdn/bdn.go]: Yes, I will remove that part and rebase on #553

**Review by AnomalRoil — APPROVED** [review]

**Pull request merged**